ui/src/utils/index.ts

```diff
@@ -11,106 +11,106 @@ const { message } = createDiscreteApi(['message']);
-    let minuteLabel = t('Minute');
+  let minuteLabel = t('Minute');
 
-    if (item > 1) {
-        minuteLabel = t('Minutes');
-    }
+  if (item > 1) {
+    minuteLabel = t('Minutes');
+  }
 
-    return `${item} ${minuteLabel}`;
+  return `${item} ${minuteLabel}`;
 };
 
 export const writeBufferToTerminal = (
-    enableZmodem: boolean,
-    zmodemStatus: boolean,
-    terminal: Terminal | null,
-    data: any
+  enableZmodem: boolean,
+  zmodemStatus: boolean,
+  terminal: Terminal | null,
+  data: any
 ) => {
-    if (!enableZmodem && zmodemStatus) return message.error('未开启 Zmodem 且当前在 Zmodem 状态, 不允许显示');
+  if (!enableZmodem && zmodemStatus) return message.error('未开启 Zmodem 且当前在 Zmodem 状态, 不允许显示');
 
-    terminal && terminal.write(new Uint8Array(data));
+  terminal && terminal.write(new Uint8Array(data));
 };
 
 export const preprocessInput = (data: string, config: ILunaConfig) => {
-    // 如果配置项 backspaceAsCtrlH 启用（值为 "1"），并且输入数据包含删除键的 ASCII 码 (AsciiDel，即 127)，
-    // 它会将其替换为退格键的 ASCII 码 (AsciiBackspace，即 8)
-    if (config.backspaceAsCtrlH === '1') {
-        if (data.charCodeAt(0) === AsciiDel) {
-            data = String.fromCharCode(AsciiBackspace);
-        }
+  // 如果配置项 backspaceAsCtrlH 启用（值为 "1"），并且输入数据包含删除键的 ASCII 码 (AsciiDel，即 127)，
+  // 它会将其替换为退格键的 ASCII 码 (AsciiBackspace，即 8)
+  if (config.backspaceAsCtrlH === '1') {
+    if (data.charCodeAt(0) === AsciiDel) {
+      data = String.fromCharCode(AsciiBackspace);
     }
+  }
 
-    // 如果配置项 ctrlCAsCtrlZ 启用（值为 "1"），并且输入数据包含 Ctrl+C 的 ASCII 码 (AsciiCtrlC，即 3)，
-    // 它会将其替换为 Ctrl+Z 的 ASCII 码 (AsciiCtrlZ，即 26)。
-    if (config.ctrlCAsCtrlZ === '1') {
-        if (data.charCodeAt(0) === AsciiCtrlC) {
-            data = String.fromCharCode(AsciiCtrlZ);
-        }
+  // 如果配置项 ctrlCAsCtrlZ 启用（值为 "1"），并且输入数据包含 Ctrl+C 的 ASCII 码 (AsciiCtrlC，即 3)，
+  // 它会将其替换为 Ctrl+Z 的 ASCII 码 (AsciiCtrlZ，即 26)。
+  if (config.ctrlCAsCtrlZ === '1') {
+    if (data.charCodeAt(0) === AsciiCtrlC) {
+      data = String.fromCharCode(AsciiCtrlZ);
     }
+  }
 
-    if (data === '^C') {
-        data = '\x03';
-    }
+  if (data === '^C') {
+    data = '\x03';
+  }
 
-    return data;
+  return data;
 };
 
 export const getOs = () => {
-    const { platform, userAgent } = navigator;
+  const { platform, userAgent } = navigator;
 
-    if (platform.startsWith('Win') || userAgent.includes('Windows')) {
-        return 'Windows';
-    }
+  if (platform.startsWith('Win') || userAgent.includes('Windows')) {
+    return 'Windows';
+  }
 
-    if (platform.startsWith('Mac') || userAgent.includes('Macintosh')) {
-        return 'macOS';
-    }
+  if (platform.startsWith('Mac') || userAgent.includes('Macintosh')) {
+    return 'macOS';
+  }
 
-    if (platform.startsWith('Linux') || userAgent.includes('Linux')) {
-        return 'Linux';
-    }
+  if (platform.startsWith('Linux') || userAgent.includes('Linux')) {
+    return 'Linux';
+  }
 };
```

Comment:
This is not a coding issue report; it appears to be an entire set of TypeScript functions related to various use cases like event broadcasting, formatting strings into smaller components and translating them from one language to another. The content you've given covers multiple areas such as error handling, UI interactions via events, utility methods for working with data types, etc.

However, no direct JavaScript errors have been reported. For instance:

```typescript
  // ...more code...
error: SyntaxError: Unexpected token U at position 54

    // This line also has some issues regarding the expected format
```

For more specific problems on this page please specify further details about what exactly needs checking.